src/NServiceBus.Transport.RabbitMQ.TransportTests/ConfigureRabbitMQTransportInfrastructure.cs

```diff
@@ -55,7 +46,38 @@ public Task Cleanup()
 
     void PurgeQueues()
     {
-        var connectionFactory = CreateConnectionFactory();
+        if (connectionStringBuilder == null)
+        {
+            return;
+        }
+
+        var connectionFactory = new ConnectionFactory
+        {
+            AutomaticRecoveryEnabled = true,
+            UseBackgroundThreadsForIO = true
+        };
+
+        object value;
+
+        if (connectionStringBuilder.TryGetValue("username", out value))
+        {
+            connectionFactory.UserName = value?.ToString();
```

Comment:
I don't mind removing the `?` since it makes sense that the value can never be null if it exists (there's no way to represent a null value in the connection string).

I think I'd rather leave the `ToString` calls. `string.ToString()` just returns the current instance, so it's cheaper than a cast and it doesn't rely on knowledge of the internals of `DbConnectionStringBuilder`.